From 704681a4cf2a5b97f9fba5da18f5c2443b2bc919 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 27 Jun 2017 10:03:44 +0800
Subject: [PATCH 2007/5242] MLK-15287 clk: imx: imx8mq: increase NOC clock
 speed

commit  f75a08c81b0a7a91c3f76aa2a5cb79402b30601e from
https://source.codeaurora.org/external/imx/linux-imx.git

NOC clock by default is running @400MHz, to achieve
best DDR access performance, increase it to 800MHz.

With CPU @1.2GHz, we can see stream copy performance
increase 24% if NOC clock is increased from 400MHz
to 800MHz.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8mq.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/imx/clk-imx8mq.c b/drivers/clk/imx/clk-imx8mq.c
index 333fc11..07d2b65 100644
--- a/drivers/clk/imx/clk-imx8mq.c
+++ b/drivers/clk/imx/clk-imx8mq.c
@@ -806,6 +806,9 @@ static void __init imx8mq_clocks_init(struct device_node *ccm_node)
 	clk_set_parent(clks[IMX8MQ_CLK_AUDIO_AHB_SRC], clks[IMX8MQ_SYS2_PLL_1000M]);
 	clk_set_rate(clks[IMX8MQ_CLK_AUDIO_AHB_DIV], 500000000);
 
+	/* increase NOC clock to achieve best DDR access performance */
+	clk_set_rate(clks[IMX8MQ_CLK_NOC_DIV], clk_get_rate(clks[IMX8MQ_SYS1_PLL_800M]));
+
 	pr_info("i.MX8MQ clock driver init done\n");
 }
 
-- 
1.7.9.5

